src/main/scala/firrtl_interpreter/InterpretiveTester.scala

```diff
@@ -49,6 +49,7 @@ class InterpretiveTester(input: String, optionsManager: HasInterpreterSuite = ne
   private var failureTime = -1L
   private var failCode: Option[Int] = None
   def fail(code: Int): Unit = {
+    interpreter.circuitState.writeVCD()
```

Comment:
Do exceptions bypass this? Wouldn't it be better to have exceptions bubble up to where it would be logical to call fail()?

Reply:
In principal I agree, but trying to make this more consistent would be a lot more work.  fail() was put into handling the most common exception that occur when bad values are poked into things but not on cycle and peek.  It did not cover the firrtl stop() node, or that the lower level FirrtlTerp is sometimes used instead, by the repl and a couple of other places.  I plan to clean up the error handling a bit in the work I am doing on the fast interpreter 2.0